### ISSUE TITLE
Support the $$ typed splice TExp operation

### ISSUE THREAD

**Issue by MHova** [ndmitchell/hlint#186]
https://ghc.haskell.org/trac/ghc/wiki/TemplateHaskell/Typed

Hlint currently complains with a `Warning: Parse error: $$` when it encounters `$$`.

**Comment by ndmitchell** [comment]
Thanks, an issue in the underlying haskell-src-exts library, which I've raised upstream: https://github.com/haskell-suite/haskell-src-exts/issues/278

**Comment by MHova** [comment]
Thanks!

**Comment by tfausak** [comment]
It seems like HSE still can't handle typed splices. I ran into the problem another way. Given the following Haskell file:

``` hs
{-# LANGUAGE TemplateHaskell #-}
x = f $$(s)
```

`hlint` 2.0.9 reports:

```
Example.hs:1:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Why not remove it.

Example.hs:2:9: Warning: Redundant bracket
Found:
  (ts)
Why not:
  ts

2 hints
```

Which tells me that HSE is parsing `$$` as an operator rather than a typed splice.

**Comment by unhammer** [comment]
https://github.com/haskell-suite/haskell-src-exts/issues/278 was closed in https://github.com/haskell-suite/haskell-src-exts/pull/432 which was merged into haskell-src-exts 1.22.0 according to https://hackage.haskell.org/package/haskell-src-exts-1.23.0/changelog . The issue seems to go away by putting e.g.
```
- haskell-src-exts-1.23.0
- hlint-2.2.7
```
into stack.yaml

**Comment by ndmitchell** [comment]
Cool - thanks for letting me know @unhammer
